### PR TITLE
Parse opushead

### PR DESCRIPTION
**Pull request by ihorserba**

### CODE REVIEW

File: Concentus.Oggfile/OpusHeader.cs [hunk 1]
```diff
@@ -11,10 +11,35 @@ internal class OpusHeader
         byte version;
         byte channel_count;
         ushort pre_skip;
-        uint input_sample_rate;
+        internal uint input_sample_rate;
```
Comment (vocoded) [inline Concentus.Oggfile/OpusHeader.cs:14]: internal is the default afaik

File: Concentus.Oggfile/OpusHeader.cs [hunk 2]
```diff
@@ -11,10 +11,35 @@ internal class OpusHeader
...
+    
+        internal static OpusHeader ParsePacket(byte[] packet, int packetLength)
+        {
+            if (packetLength < 19)
+                return null;
+
+            if (!"OpusHead".Equals(Encoding.UTF8.GetString(packet, 0, 8)))
+                return null;
+
+            OpusHeader returnVal = new OpusHeader();
```
Comment (vocoded) [inline Concentus.Oggfile/OpusHeader.cs]: can we call this `header`?

File: Concentus.Oggfile/OpusOggReadStream.cs [hunk 3]
```diff
@@ -57,6 +59,17 @@ public OpusTags Tags
...
         }
 
+        /// <summary>
+        /// Gets the input sample rate from the header from the OpusHead Ogg packet.
+        /// </summary>
+        public uint InputSampleRate
+        {
+            get
+            {
+                return _header.input_sample_rate;
```
Comment (vocoded) [inline Concentus.Oggfile/OpusOggReadStream.cs:72]: this can be null, need to guard

File: Concentus.Oggfile/OpusOggReadStream.cs [hunk 4]
```diff
@@ -81,6 +94,28 @@ public string LastError
...
+            {
+                _endOfStream = true;
+            }
+
+            if (_endOfStream)
+            {
+                return null;
+            }
+
+            var result = _nextDataPacket;
```
Comment (vocoded) [inline Concentus.Oggfile/OpusOggReadStream.cs:117]: unused or this is a bug

File: Concentus.Oggfile/OpusHeader.cs [hunk 5]
```diff
@@ -11,10 +11,35 @@ internal class OpusHeader
...
         byte mapping_family;
         byte stream_count;
         byte coupled_count;
+    
+        internal static OpusHeader ParsePacket(byte[] packet, int packetLength)
+        {
+            if (packetLength < 19)
+                return null;
+
+            if (!"OpusHead".Equals(Encoding.UTF8.GetString(packet, 0, 8)))
```
Comment (vocoded) [inline Concentus.Oggfile/OpusHeader.cs:25]: this check is already made before the method is called

File: Concentus.Oggfile/OpusHeader.cs (hunk 2 above)
Comment (ihorserba) [reply Concentus.Oggfile/OpusHeader.cs]: Renamed

File: Concentus.Oggfile/OpusOggReadStream.cs (hunk 3 above)
Comment (ihorserba) [reply Concentus.Oggfile/OpusOggReadStream.cs:72]: added check

File: Concentus.Oggfile/OpusOggReadStream.cs (hunk 4 above)
Comment (ihorserba) [reply Concentus.Oggfile/OpusOggReadStream.cs:117]: Yes, looks like a bug. Fixed

File: Concentus.Oggfile/OpusHeader.cs (hunk 5 above)
Comment (ihorserba) [reply Concentus.Oggfile/OpusHeader.cs:25]: Yes, it's redundant a little bit. 
I've used the same style as originally has been used for OpusTags.ParsePacket(), so I'd keep it.

**Pull request merged**